src/irmin-pack/unix/lower.ml

```diff
@@ -131,7 +131,7 @@ module Make_volume (Io : Io.S) (Errs : Io_errors.S with module Io = Io) = struct
   let eq a b = identifier_eq ~id:(identifier b) a
 
   let read_range_exn ~off ~min_len ~max_len b = function
-    | Empty _ -> ()
+    | Empty _ -> assert (min_len = 0)
```

Comment:
I believe it's an assert false, since we never request a `min_len = 0`. I was debugging a read that produced a weird buffer contents later, and stumbled upon this unrelated code: If the volume is empty, then one shouldn't read from it? But if no error is signaled, then what random values will we find in [b] after?

Anyway I think it's dead code, but would you prefer an `Error Read_out_of_bounds`?

Reply:
Mm, I see. I don't in practice expect this to get hit which I think is why I just returned unit (ie, no data read from an empty volume), but it might be wise to raise an exception (I like `Read_out_of_bounds` but maybe with some context so we know it came from here) to encode our assumption that it shouldn't happen.

Reply:
(Good catch, btw! I do think it is correct to raise an exception if we expect this to not happen.)